src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -235,6 +235,7 @@ public virtual Task DispatchEventAsync(ulong eventHandlerId, EventFieldInfo fiel
             catch (Exception e)
             {
                 HandleException(e);
+                return Task.CompletedTask;
```

Comment:
This is a bit of a nitpick, but I'd be tempted to replace lines 237 and 238 with:

```cs
// Treat synchronous exceptions the same as async ones
task = Task.FromException(ex);
```

... and then continue. It means we have fewer code paths to reason about. I think the behavior would be the same.

Do you have any view on this? Totally fine to do it whichever way you prefer, of course!

Reply:
I know my version would allocate more. I'm not super worried about that since this is an error path anyway, but it's up to you.

Reply:
I'm curious, what's the advantage other than it goes through the same code path. I feel this is more straightforward, but I want to know if there's something I'm missing.